azure-pipelines/main-pipeline.yml

```diff
@@ -253,15 +263,100 @@ stages:
               # Sonar Cloud Analysis
 
               - task: SonarCloudAnalyze@1
+                displayName: Analyze
 
               - task: SonarCloudPublish@1
                 inputs:
                   pollingTimeoutSec: "300"
+                displayName: Publish Sonar Results
+
+              # Docker Image is created and pushed to ACR with build specific tag.
+
+              - template: "./azure-templates/docker-template.yml"
+                parameters:
+                  repository: ${{ service.name }}
+                  tag: "$(Build.BuildNumber)"
+                  Dockerfile: ./${{ service.name }}/Dockerfile
+                  containerRegistry: "acr-team3"
+              
+              # Docker Image is created and pushed to ACR with latest tag.
+
+              - template: "./azure-templates/docker-template.yml"
+                parameters:
+                  repository: ${{ service.name }}
+                  tag: "latest"
+                  Dockerfile: ./${{ service.name }}/Dockerfile
+                  containerRegistry: "acr-team3"
+
+  # Deployment Stage. Using images pushed to the Azure Container Registry, and a helm Chart.
+
+  - stage: Deploy
+    displayName: Deploy to AKS Cluster
+    dependsOn: Check
+    condition: and(succeeded(), in(variables['Build.Reason'], 'IndividualCI', 'BatchedCI'))
```

Comment:
we usually like Manual builds, and for that matter every type of build besides PullRequest, to be deployed.

Reply:
Addressed in upcoming pr.